### PR TITLE
mixed precision design doc

### PR DESCRIPTION
**Pull request by dzhwinter**
preview
https://github.com/PaddlePaddle/Paddle/blob/58cdf8a8d58e6b619907de109663e5cf18d82767/doc/survey/mixed_precision.md

### CODE REVIEW

File: doc/survey/mixed_precision.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+```python
+  >>> import paddle.fluid as fluid
+
+  # 1. set datatype to float16
+  >>> data = fluid.layers.data(name="data", shape=[32, 32], dtype="float16") # set the type to float16
+  >>> labels = fluid.layers.data(name="label", shape=[1], dtype="float16") # set the type to float16
+  >>> fc = fluid.layers.fc(input=data, size=1000, act="tanh")
+  
+  # 2. use batch norm (auto insert cast op when enter/exit)
+  >>> with fluid.dtype("float32"):
```
Comment (panyx0718) [inline doc/survey/mixed_precision.md:95]: why not
```python
x = fluid.cast(x, 'fluid32')
x = fluid.bn(x)
x = fluid.cast(x, 'float16')
```

File: doc/survey/mixed_precision.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  
+  >>> out = layers.assign(hidden, fc)
+  >>> loss = layers.mean(out)
+
+  # 3. use float16 optimizer(a wrapper to parameters)
+  >>> adam = fluid.optimizer.MixedOptimizer(fluid.optimizer.Adam())
+  >>> adam.minimize(loss)
+	
+  # 4. save/convert fp16 model to fp32
+  >>> def tofp16(program):
```
Comment (panyx0718) [inline doc/survey/mixed_precision.md:106]: Is this needed? What does it do?

**Review by panyx0718 — APPROVED** [review]

**Comment by shanyi15** [comment]
hi，doc目录下的东西目前已全部在FluidDoc repo管理，请不要再往Paddle repo此目录下merge，谢谢！
@dzhwinter
